In [1]:
from transformers import AutoModel, AutoTokenizer
from sentence_transformers import SentenceTransformer

First, we are going to use a BERT-like model to create embeddings for words.

In [2]:
tokenizer = AutoTokenizer.from_pretrained("microsoft/deberta-base")
model = AutoModel.from_pretrained("microsoft/deberta-v3-xsmall")

/usr/local/lib/python3.12/dist-packages/huggingface_hub/utils/_auth.py:94: UserWarning: 
The secret `HF_TOKEN` does not exist in your Colab secrets.
To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.
You will be able to reuse this secret in all of your notebooks.
Please note that authentication is recommended but still optional to access public models or datasets.
  warnings.warn(


Generating the tokens.

In [3]:
tokens = tokenizer('hello world', return_tensors='pt')
output = model(**tokens)[0]

output.shape

torch.Size([1, 4, 384])

Here, each token was embedded into a tensor of dimension 384. Note that the final embeddings contain contextual information of their neighboors, as they are the outputs from a BERT model.

We can also generate sentence embeddings, in similar fashion.

In [4]:
model = SentenceTransformer("sentence-transformers/all-mpnet-base-v2")
output = model.encode("sentence embedding")

output.shape

(768,)

Here, the sentence was embedded into a 768-shaped tensor.